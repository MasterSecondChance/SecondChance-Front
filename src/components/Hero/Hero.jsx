import React from 'react';
import './Hero.scss';

const Hero = ({ title, image }) => {
  return (
    <section className="Hero">
      <div>
        <div className="Hero__circle"></div>
        <h1 className="Hero__title" tabindex="0">{title}</h1>
        <figure className="Hero__imageContainer">
          <img className="Hero__image" src={image} width="440" role="presentation" />
        </figure>
      </div>
    </section>
  )
}

export default Hero