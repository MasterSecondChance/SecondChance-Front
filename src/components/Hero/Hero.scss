@import '../../assets/styles/containers/Variables.scss';

.Hero {
  padding: $spacing-03 $spacing-02;
  font-size: 1.5rem;

  &__circle {
    display: none;
  }

  &__title {
    margin: 0;
  }

  &__imageContainer {
    display: none;
  }
}

@media screen and (min-width: 768px) {
  .Hero {
    height: 220px;
    padding: 80px $spacing-03 0;
    overflow: hidden;
    font-size: 2.5rem;
    background: $primary-color-dark;
    background: linear-gradient(93deg, $primary-color-dark 0%, $primary-color 100%);
    filter: progid:DXImageTransform.Microsoft.gradient(startColorstr="#b38dc9", endColorstr="#c7a0da", GradientType=1);

    &>div {
      position: relative;
      max-width: 1000px;
      height: inherit;
      margin: 0 auto;
    }

    &__circle {
      position: absolute;
      top: -150%;
      left: 2%;
      display: block;
      width: 400px;
      height: 400px;
      border: 1px solid $primary-color-light-2;
      border-radius: 50%;
    }

    &__title {
      position: relative;
      z-index: 2;
    }

    &__imageContainer {
      position: absolute;
      right: 0;
      bottom: 0;
      display: block;
      margin: 0;
    }

    &__image {
      max-width: 100%;
    }
  }
}