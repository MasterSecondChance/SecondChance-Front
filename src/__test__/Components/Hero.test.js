import React from 'react';
import { mount, shallow } from 'enzyme';
import ProviderMock from '../../__mocks__/providerMock';
import Hero from '../../components/Hero/Hero';
import renderer, { create } from 'react-test-renderer';

describe('<Hero />', () => {
  test('Render del componente Hero', () => {
    const hero = shallow(
      <ProviderMock>
        <Hero />
      </ProviderMock>
    );
    expect(hero.length).toEqual(1);
  })
  test('Render del titulo en Hero', () => {
    const hero = shallow(
      <Hero title={'Trode'} />
    );
    expect(hero.find('.Hero__title').text()).toEqual('Trode')
  })
  test('Render de Imagen', () => {
    const hero = shallow(<Hero image={'logo.svg'} />);
    const img = hero.find('img');
    const src = img.prop('src');
    expect(src).toBe('logo.svg');
  })
})

describe('Hero Snapshot', () => {
  test('Comprobar la UI del Hero', () => {
    const hero = create(<Hero />)
    expect(hero.toJSON()).toMatchSnapshot();
  })
})