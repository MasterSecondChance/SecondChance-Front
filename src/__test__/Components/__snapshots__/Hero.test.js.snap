// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Snapshot Comprobar la UI del Hero 1`] = `
<section
  className="Hero"
>
  <div>
    <div
      className="Hero__circle"
    />
    <h1
      className="Hero__title"
      tabIndex="0"
    />
    <figure
      className="Hero__imageContainer"
    >
      <img
        className="Hero__image"
        role="presentation"
        width="440"
      />
    </figure>
  </div>
</section>
`;
